crates/extricrate/src/lib.rs

```diff
@@ -183,45 +222,43 @@ pub mod dependencies {
         let src_folder = entry_point
             .parent()
             .expect("Failed to get entry point parent folder");
-        files_to_visit.push_back((entry_point.clone(), "main".to_owned()));
+        files_to_visit.push_back(FileToVisit {
+            file: entry_point.clone(),
+            module_ancestors: vec![],
+        });
         while let Some(file_to_visit) = files_to_visit.pop_front() {
-            if files_visited.contains(&file_to_visit) {
+            if files_visited.contains(&file_to_visit.file) {
                 continue;
             }
 
-            if !file_to_visit.0.exists() {
+            if !file_to_visit.file.exists() {
                 return Err(ListUseStatementError::FileNotFound);
             }
 
-            let content = read_to_string(&file_to_visit.0)
+            let content = read_to_string(&file_to_visit.file)
                 .map_err(|_| ListUseStatementError::FileNotReadable)?;
 
             let parsed_file =
                 parse_file(&content).map_err(|_| ListUseStatementError::FileNotParsable)?;
 
-            let mut visitor = UseVisitor::new();
+            let mut visitor = Visitor::new();
             visitor.visit_file(&parsed_file);
-            for statement in visitor
-                .statements
-                .iter()
-                .flat_map(|dependency| &dependency.items)
-            {
-                match &statement.statement_type {
-                    UseStatementType::Simple(name) => {
-                        if is_local_import(&statement.module_name) {
-                            let file_to_visit = get_path_from_module_name(src_folder, name).ok_or(
-                                ListUseStatementError::ModuleDoesNotExists(name.to_string()),
-                            )?;
-                            files_to_visit.push_back((file_to_visit, name.to_owned()));
-                        }
-                    }
-                    UseStatementType::Alias(name, _) => todo!(),
-                    UseStatementType::WildCard => todo!(),
-                }
-            }
 
+            files_to_visit.extend(visitor.mod_statements.iter().filter_map(|s| {
```

Comment:
Unless I'm mistaken, here you're silently swallowing errors (could not find module). I think it'd be better to propagate them.